### PR TITLE
Glimpse2 concordance

### PR DESCRIPTION
**Pull request by LouisLeNezet**
## PR checklist

Final module of GLIMPSE2.

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/glimpse2/concordance/main.nf [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    label 'process_low'
+
+    conda "bioconda::glimpse-bio=2.0.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/glimpse-bio:2.0.0--hf340a29_0':
+        'quay.io/biocontainers/glimpse-bio:2.0.0--hf340a29_0' }"
+
+    input:
+    tuple val(meta), path(estimate), path(estimate_index), path(truth), path(truth_index), path(freq), path(freq_index), path(samples), val(region)
+    tuple val(meta2), path(groups), val(bins), val(ac_bins), val(allele_counts)
```
Comment (nvnieuwk) [inline modules/nf-core/glimpse2/concordance/main.nf:12]: This is bit of a bummer situation but I don't see a better solution indeed, good job!

File: modules/nf-core/glimpse2/concordance/main.nf [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    val(min_val_gl)
+    val(min_val_dp)
+
+    output:
+    tuple val(meta), path("*.error.*.txt.gz") , emit: errors
+    tuple val(meta), path("*.rsquare.*.txt.gz"), emit: rsquare
```
Comment (nvnieuwk) [inline modules/nf-core/glimpse2/concordance/main.nf:18]: Can you make a separate output for each `errors` or `rsquare` file created? It's usually easier to merge them than to split them on file name for example :)

File: modules/nf-core/glimpse2/concordance/main.nf [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    script:
+    def args         = task.ext.args   ?: ''
+    def prefix       = task.ext.prefix ?: "${meta.id}"
+    def samples_cmd  = samples         ? "--samples ${samples}"             : ""
+    def groups_cmd   = groups          ? "--groups ${groups}"               : ""
+    def bins_cmd     = bins            ? "--bins ${bins}"                   : ""
+    def ac_bins_cmd  = ac_bins         ? "--ac-bins ${ac_bins}"             : ""
+    def ale_ct_cmd   = allele_counts   ? "--allele-counts ${allele_counts}" : ""
+
+    if (((groups ? 1:0) + (bins ? 1:0) + (ac_bins ? 1:0) + (allele_counts ? 1:0)) != 1) error "One and only one argument should be selected between groups, bins, ac_bins, allele_counts"
```
Comment (nvnieuwk) [inline modules/nf-core/glimpse2/concordance/main.nf:33]: Smart!

File: modules/nf-core/glimpse2/concordance/meta.yml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      pattern: "*.{vcf,bcf,vcf.gz,bcf.gz}"
+
+  - truth:
+      type: file
+      description: Validation dataset called at the same positions as the imputed file.
+      pattern: "*.{vcf,bcf,vcf.gz,bcf.gz}"
+
+  - estimate:
+      type: file
+      description: Imputed data.
```
Comment (nvnieuwk) [inline modules/nf-core/glimpse2/concordance/meta.yml:41]: Can you make this description a bit more clear?

File: modules/nf-core/glimpse2/concordance/meta.yml [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+  - samples:
+      type: file
+      description: List of samples to process, one sample ID per line.
+      pattern: "*.{txt,tsv}"
+
+  - groups:
+      type: file
+      description: Alternative to frequency bins, group bins are user defined, provided in a file.
+      pattern: "*.{txt,tsv}"
```
Comment (nvnieuwk) [inline modules/nf-core/glimpse2/concordance/meta.yml:52]: Can you add the other inputs (`bins`, `ac_bins` and `allele_counts` too?)

**Review by nvnieuwk — CHANGES_REQUESTED** [review]
Mainly some small suggestions to meta and outputs, otherwise very good job this didn't look like an easy one :)

File: modules/nf-core/glimpse2/concordance/meta.yml [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+  - min_val_gl:
+      type: float
+      description: |
+        Minimum genotype likelihood probability P(G|R) in validation data.
+        Set to zero to have no filter of if using –gt-validation
+
+  - min_val_dp:
+      type: integer
+      description: |
+        Minimum coverage in validation data.
```
Comment (matthdsm) [inline modules/nf-core/glimpse2/concordance/meta.yml:82]:
```suggestion
        Minimum coverage in validation data.
```

**Pull request merged**